File: MichiMark/Domain/Model/MemberDomain.swift
import Foundation

struct MemberDomain: Equatable, Sendable {
    let id: MemberID
    var memberName: String/// 入力必須（Stateでは空欄許可）
    var mailAddress: String?/// 将来拡張（共有・招待・通知など）
    var isVisible: Bool/// True: 表示 / False: 非表示（初期値は true 想定）

    var isDeleted: Bool/// 論理削除
    let createdAt: Date/// 登録日（初回のみ設定）
    var updatedAt: Date/// 更新日（保存時更新）

    init(
        id: MemberID,
        memberName: String,
        mailAddress: String? = nil,
        isVisible: Bool = true,
        isDeleted: Bool = false,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.memberName = memberName
        self.mailAddress = mailAddress
        self.isVisible = isVisible
        self.isDeleted = isDeleted
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: flutter/lib/domain/master/member/member_domain.dart
import 'package:equatable/equatable.dart';

class MemberDomain extends Equatable {
  final String id;

  /// 表示名（入力必須）
  final String memberName;

  /// メールアドレス（将来拡張：共有・招待・通知など）
  final String? mailAddress;

  /// 表示フラグ（true: 表示 / false: 非表示）
  final bool isVisible;

  /// 論理削除フラグ
  final bool isDeleted;

  /// 登録日（初回のみ設定）
  final DateTime createdAt;

  /// 更新日（保存時更新）
  final DateTime updatedAt;

  const MemberDomain({
    required this.id,
    required this.memberName,
    this.mailAddress,
    this.isVisible = true,
    this.isDeleted = false,
    required this.createdAt,
    required this.updatedAt,
  });

  MemberDomain copyWith({
    String? id,
    String? memberName,
    String? mailAddress,
    bool? isVisible,
    bool? isDeleted,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return MemberDomain(
      id: id ?? this.id,
      memberName: memberName ?? this.memberName,
      mailAddress: mailAddress ?? this.mailAddress,
      isVisible: isVisible ?? this.isVisible,
      isDeleted: isDeleted ?? this.isDeleted,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  List<Object?> get props => [
        id,
        memberName,
        mailAddress,
        isVisible,
        isDeleted,
        createdAt,
        updatedAt,
      ];
}
